server/neptune/workflows/internal/pr/revision/processor.go

```diff
@@ -68,6 +69,14 @@ func (p *Processor) Process(ctx workflow.Context, prRevision Revision) {
 		futures = append(futures, future)
 	}
 
+	// Mark checkruns as aborted if the context was cancelled, this typically happens if revisions arrive in quick succession
+	defer func() {
+		if temporal.IsCanceledError(ctx.Err()) {
+			ctx, _ := workflow.NewDisconnectedContext(ctx)
+			p.markCheckRunsAborted(ctx, prRevision, roots)
+		}
+	}()
```

Comment:
```suggestion
	defer func() {
		if temporal.IsCanceledError(ctx.Err()) {
			ctx, _ := workflow.NewDisconnectedContext(ctx)
			p.markCheckRunsAborted(ctx, prRevision, roots)
			return
		}
		// At this point, all workflows should be successful, and we can mark combined plan check run as success
	p.markCombinedCheckRun(ctx, prRevision, github.CheckRunSuccess, "")
	}()
```

Looks good, one nit: can you move this up here